src/main/java/org/spongepowered/api/entity/living/Living.java

```diff
@@ -287,4 +288,20 @@
      * @param invisible Whether this entity is invisible or not
      */
     void setInvisible(boolean invisible);
+
+    /**
+     * Returns whether this living entity is invisible to a given player.
+     *
+     * @param otherPlayer The player to toggle visibility of this entity for
+     * @return Whether this entity is invisible to the given player
+     */
+    boolean isInvisible(Player otherPlayer);
+
+    /**
+     * Sets whether this living entity is invisible to a given player.
+     *
+     * @param otherPlayer The player to toggle visibility of this entity for
+     * @param invisible Whether this entity is invisible to the given player
+     */
+    void setInvisible(Player otherPlayer, boolean invisible);
```

Comment:
There's no real reason to restrict it to `Human` or `Player`, and additionally, `Living` holds the existing invisibility methods.